### PR TITLE
`ch-convert`: allow empty dirs as output

### LINKED ISSUE #1612: `ch-convert`: accept empty directory as valid output path
This helps a lot in environments where the permission on the output path must be set by root or a different user.

For starters:
- `dir_out_validate()` should accept empty directories 
- `cv_*_dir()` functions should be modified to work with an existing path

### PR DESCRIPTION
**Pull request by lucaudill**
closes #1612.

### CODE REVIEW

**Comment by lucaudill** [comment]
@MFTabriz the feature seems to be working on my end, do you want to try it out and let me know what you think?

**Comment by MFTabriz** [comment]
@lucaudill Thanks a lot for your patch. I’ll go through the changes and also test them this weekend.

**Comment by MFTabriz** [comment]
@lucaudill Thanks for taking care of it. From my simple local tests, it’s working fine. 
About `unsquashfs`, I think we should open an issue for improving their documentation.

**Comment by lucaudill** [comment]
@MFTabriz Awesome! Thanks for testing it out for me.

File: test/run/ch-convert.bats [hunk 1]
```diff
@@ -383,6 +388,67 @@ test_from () {
...
+    printf 'FROM alpine:3.17\n' | podman_ build -t tmpimg -
+
+    # squash
+    touch "${BATS_TMPDIR}/tmpimg.sqfs"
+    ch-convert -i ch-image -o squash tmpimg "$BATS_TMPDIR/tmpimg.sqfs"
+
+    # tar
+    ch-convert -i ch-image -o tar tmpimg "$BATS_TMPDIR/tmpimg.tar.gz"
+
+
```
Comment (reidpr) [inline test/run/ch-convert.bats:412]:
```suggestion
```

**Review by reidpr — CHANGES_REQUESTED** [review]
Looks nice! Couple of minor nits in-line.

**Review by reidpr — APPROVED** [review]

**Pull request merged**